### PR TITLE
cp+LibCore: Implement the granular `--preserve` flag

### PR DESCRIPTION
**Pull request by timschumi**
(Hopefully) one less port that is needed for self-hosting.

### CODE REVIEW

**Comment by timschumi** [comment]
I'm not too sure whether how LibCore handles file and directory permissions when copying is compliant with the expected behavior, but a fix for that should rather be in a separate PR, as this tries to not change the resulting behavior at all.

File: Userland/Libraries/LibCore/File.h [hunk 1]
```diff
@@ -55,9 +55,11 @@ class File final : public IODevice {
         No,
     };
 
-    enum class PreserveMode {
-        Nothing,
-        PermissionsOwnershipTimestamps,
+    enum PreserveMode {
```
Comment (linusg) [inline Userland/Libraries/LibCore/File.h:58]: Plain `enum`? :catdogalert: Pretty sure you want to use `AK_ENUM_BITWISE_OPERATORS` instead.

**Review by linusg — CHANGES_REQUESTED** [review]
https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md
> Don't use specific component names, e.g. C++ class names, as the category either - mention them in the summary instead. E.g. `LibGUI: Brief description of what's being changed in FooWidget` rather than `FooWidget: Brief description of what's being changed`

(It used to say something different a long time ago, I know)
`LibCore/ArgsParser: ` is fine, or just regular `LibCore: Do stuff in ArgsParser`.

File: Userland/Libraries/LibCore/File.h (hunk 1 above)
Comment (timschumi) [reply Userland/Libraries/LibCore/File.h]: That's what I was looking for but could not find. 🙃 

The code looks much better now, and the commits have the correct category. :^)

**Review by linusg — APPROVED** [review]

**Pull request merged**